File: ios/MFDeviceInfo.m
//
//  MFDeviceInfo.m
//  AlamiReactCart
//
//  Created by Moch Fariz Al Hazmi on 20/09/22.
//

#import <Foundation/Foundation.h>

#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(MFDeviceInfo, NSObject)

RCT_EXTERN_METHOD(getDeviceID:(RCTResponseSenderBlock)callback)

@end

File: ios/MFDeviceInfo.swift
//
//  MFDeviceInfo.swift
//  AlamiReactCart
//
//  Created by Moch Fariz Al Hazmi on 20/09/22.
//

import Foundation

@objc(MFDeviceInfo)
class MFDeviceInfo: NSObject {
  
  private var deviceID = "";
  
  @objc
  func getDeviceID(_ callback: RCTResponseSenderBlock) {
    if let identifierForVendor = UIDevice.current.identifierForVendor {
      deviceID = identifierForVendor.uuidString
      callback([deviceID])
    }
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  func constantsToExport() -> [String: Any] {
    return ["initialDeviceID": ""]
  }
  
}
